packages/vaex-contrib/vaex/contrib/io/gbq.py

```diff
@@ -0,0 +1,194 @@
+        where species = "virginica"
+    """
+
+    >>> df = query_google_big_query(query=query)
+    >>> df.head(3)
+    #    sepal_length    sepal_width    petal_length    petal_width  species
+    0             4.9            2.5             4.5            1.7  virginica
+    1             5.7            2.5             5              2    virginica
+    2             6              2.2             5              1.5  virginica
+
+    '''
+    client = google.cloud.bigquery.Client(project=client_project, credentials=credentials)
+    job = client.query(query=query)
+    return vaex.from_arrow_table(job.to_arrow())
+
+
+def download_google_bigquery_table(project, dataset, table, columns=None, condition=None, export=None, client_project=None, credentials=None):
+    '''Download (stream) an entire Google BigQuery table locally.
+
+    :param str project: The Google BigQuery project that owns the table.
+    :param str dataset: The dataset the table is part of.
+    :param str table: The name of the table
+    :param list columns: A list of columns (field names) to download. If None, all columns will be downloaded.
+    :param str condition: SQL text filtering statement, similar to a WHERE clause in a query. Aggregates are not supported.
+    :param str export: Pass an filename or path to download the table as an Apache Arrow file, and leverage memory mapping. If `None` the DataFrame is in memory.
+    :param str client_project: The ID of the project that executes the query. Will be passed when creating a job. If `None`, it will be set with the same value as `project`.
+    :param credentials: The authorization credentials to attach to requests. See google.auth.credentials.Credentials for more details.
+    :rtype: DataFrame
+
+    Example:
+
+    >>> import os
+    >>> os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '../path/to/project_access_key.json'
+    >>> from vaex.contrib.io.gbq import download_google_bigquery_table
+
+    >>> client_project = 'my_project_id'
+    >>> project = 'bigquery-public-data'
+    >>> dataset = 'ml_datasets'
+    >>> table = 'iris'
+    >>> columns = ['species', 'sepal_width', 'petal_width']
+    >>> conditions = 'species = "virginica"'
+    >>> df = download_google_bigquery_table(project=project,
+                                            dataset=dataset,
+                                            table=table,
+                                            columns=columns,
+                                            condition=conditions,
+                                            client_project=client_project)
+    >>> df.head(3)
+    #    sepal_width    petal_width  species
+    0            2.5            1.7  virginica
+    1            2.5            2    virginica
+    2            2.2            1.5  virginica
+    >>>
+
+    '''
+    # Instantiate the table path and the reading session
+    bq_table = f'projects/{project}/datasets/{dataset}/tables/{table}'
```

Comment:
What about making this a parameter, like table_path_template='projects/{project}/datasets/{dataset}/tables/{table}' (note: without the f)
and then do: 
```suggestion
    bq_table = table_path_template.format(project=project, dataset=dataset, table=table)
```
That removes some of the magic.